tokio/src/sync/semaphore.rs

```diff
@@ -73,6 +73,21 @@ use std::sync::Arc;
 /// }
 /// ```
 ///
+/// Limit access to a file:
+/// ```
+/// use std::io::{Result, Write};
+/// use std::fs::File;
```

Comment:
We want to use the async version of file to avoid blocking the thread.
```suggestion
/// use tokio::fs::File;
```
You may want to base your example on something from [this page](https://tokio.rs/tokio/tutorial/io). (You don't have to change what the example does — just do the same thing but with Tokio's file instead.)